tsml_eval/_wip/shapelets/fixed_length_STC.sh

```diff
@@ -0,0 +1,198 @@
+if ((count>=start_point)); then
+
+# This is the loop to keep from dumping everything in the queue which is maintained around max_num_submitted jobs
+num_jobs=$(squeue -u ${username} --format="%20P %5t" -r | awk '{print $2, $1}' | grep -e "R ${queue_alias}" -e "PD ${queue_alias}" | wc -l)
+while [ "${num_jobs}" -ge "${max_num_submitted}" ]
+do
+    echo Waiting 60s, "${num_jobs}" currently submitted on ${queue}, user-defined max is ${max_num_submitted}
+    sleep 60
+    num_jobs=$(squeue -u ${username} --format="%20P %5t" -r | awk '{print $2, $1}' | grep -e "R ${queue_alias}" -e "PD ${queue_alias}" | wc -l)
+done
+
+mkdir -p "${out_dir}${classifier}/${dataset}/"
+
+# This skips jobs which have test/train files already written to the results directory. Only looks for Resamples, not Folds (old file name)
+array_jobs=""
+for (( i=start_fold-1; i<max_folds; i++ ))
+do
+    if [ -f "${results_dir}${classifier}/Predictions/${dataset}/testResample${i}.csv" ]; then
+        if [ "${generate_train_files}" == "true" ] && ! [ -f "${results_dir}${classifier}/Predictions/${dataset}/trainResample${i}.csv" ]; then
+            array_jobs="${array_jobs}${array_jobs:+,}$((i + 1))"
+        fi
+    else
+        array_jobs="${array_jobs}${array_jobs:+,}$((i + 1))"
+    fi
+done
+
+if [ "${array_jobs}" != "" ]; then
+
+ # Determine if length_selector argument should be included
+if [ "$classifier" == "fixedlengthshapelettransformclassifier" ]; then
+    length_selector="--kwargs length_selector FIXED str"
+else
+    length_selector=""
+fi
+# This creates the scrip to run the job based on the info above
+echo "#!/bin/bash
+#SBATCH --mail-type=${mail}
+#SBATCH --mail-user=${mailto}
+#SBATCH -p ${queue}
+#SBATCH -t ${max_time}
+#SBATCH --job-name=${classifier}${dataset}
+#SBATCH --array=${array_jobs}
+#SBATCH --mem=${max_memory}M
+#SBATCH -o ${out_dir}${classifier}/${dataset}/%A-%a.out
+#SBATCH -e ${out_dir}${classifier}/${dataset}/%A-%a.err
+#SBATCH --nodes=1
+
+. /etc/profile
+
+module load anaconda/py3.10
+source activate $env_name
+
+# Input args to the default classification_experiments are in main method of
+# https://github.com/time-series-machine-learning/tsml-eval/blob/main/tsml_eval/experiments/classification_experiments.py
+python -u ${script_file_path} ${data_dir} ${results_dir} ${classifier} ${dataset} \$((\$SLURM_ARRAY_TASK_ID - 1)) ${generate_train_files} ${predefined_folds} ${normalise_data} ${length_selector}" > generatedFile.sub
+
+echo "${count} ${classifier}/${dataset}"
+
+sbatch < generatedFile.sub
+
+else
+    echo "${count} ${classifier}/${dataset}" has finished all required resamples, skipping
+fi
+
+fi
+done
+done < ${datasets}
+
+echo Finished submitting jobs
+
+# Wait for all jobs to finish
+echo "Waiting for all jobs to complete..."
+while squeue -u ${username} | grep -q .; do
+    echo "Some jobs are still running:"
+    squeue -u ${username}
+    echo "Waiting for jobs to complete..."
+    sleep 120
+done
+
+echo "All jobs completed."
+
```

Comment:
Currently the second script - below - doesn't work. I have tested it by not waiting for all experiments. no outpout and doesn't generate .err or .out